src/send_future.rs

```diff
@@ -193,10 +213,11 @@ where
 
         match mem::replace(&mut this.inner, SendFutureInner::Done) {
             SendFutureInner::Sending(mut send_fut) => match send_fut.poll_unpin(ctx) {
```

Comment:
I think you could use `?` here.


```suggestion
            SendFutureInner::Sending(mut send_fut) => match send_fut.poll_unpin(ctx)? {
```